io_scene_godot/converters/animation.py

```diff
@@ -34,23 +61,47 @@ def __init__(self):
         self['length'] = 0
         self.track_count = 0
 
+    def __hash__(self):
+        # magical hash value, it is safe as
+        # lru_cache is used as 'typed'
+        return 0x25161239850235
+
     def add_track(self, track):
         """add a track to animation resource"""
-        track_length = track.last_frame() / bpy.context.scene.render.fps
+        track_length = track.frame_end() / bpy.context.scene.render.fps
         if track_length > self['length']:
             self['length'] = track_length
 
         track_id_str = 'tracks/{}'.format(self.track_count)
         self.track_count += 1
 
         self[track_id_str + '/type'] = '"{}"'.format(track.type)
+        self[track_id_str + '/path'] = track.path
+        self[track_id_str + '/interp'] = track.interp
         if track.type == 'transform':
-            self[track_id_str + '/path'] = track.path
-            self[track_id_str + '/interp'] = LINEAR_INTERPOLATION
             self[track_id_str + '/keys'] = transform_frames_to_keys(
-                track.frame_begin, track.frames
+                track.frames, track.values, track.interp
+            )
+        elif track.type == 'value':
+            self[track_id_str + '/keys'] = value_frames_to_keys(
+                track.frames, track.values, track.interp
             )
 
+    @lru_cache(maxsize=32, typed=True)
```

Comment:
lru_cache is shared between instances, so this means there are 32 cache values in total. This may hide future bugs if there are more than 32 tracks in an object or if tracks are added to objects not at the same time (eg 5 tracks from one object, 50 tracks from other objects, and then another 5 from the first object). I suppose the "worst" that can happen are duplicate tracks, but that still isn't ideal.

Is there a better way to prevent this running too many times? Perhaps checking `fcurve_data_path` against a list of already exported data paths inside the function?

Reply:
emm, seems previously I misunderstood what lru_cache is doing(I thought it is caching parameters..).  Now I think this lru_cache is not really needed or shouldn't be used. A lru_cached function with no return only have the benefit to avoid duplicating function calls, but I think it wont happen, as fcurves, node_path and the partial function won't be same at the same time